### PR TITLE
Improved credential select form accessibility

### LINKED ISSUE #24412: Accessibility of 2FA method selection
### Description

On 2FA method selection it's not possible to gain focus and select, using a keyboard (Tab) on either of the options, since they are neither links, nor button (they're "div"s). This is also the case for the caret browsing mode.

I will link a MR in a short while

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by Paul-Van-Uytvinck**
Closes https://github.com/keycloak/keycloak/issues/24412

### CODE REVIEW

**Comment by ahus1** [comment]
@agagancarczyk - I know you've been into accessibility, would you mind to have a look at this issue and pull request?

**Comment by jonkoops** [comment]
Thanks for this enhancement @Paul-Van-Uytvinck, it's really important that we get the accessibility of this right. I was thinking, perhaps the better option here would be to use more semantically valid controls such as buttons that submit the form. For example:

```html
<form>
  <button type="submit" name="authenticator" value="google-authenticator">
    Google Authenticator
  </button>
  <button type="submit" name="authenticator" value="microsoft-authenticator">
    Microsoft Authenticator
  </button>
</form>
```

This will submit the form with the appropriate `value` attached to the button when clicked. This also removes the need to have any JavaScript on the page. I've created [an example](https://stackblitz.com/edit/web-platform-6xfovm?file=index.html) to see how this works, let me know what you think.

**Comment by Paul-Van-Uytvinck** [comment]
> Thanks for this enhancement @Paul-Van-Uytvinck, it's really important that we get the accessibility of this right. I was thinking, perhaps the better option here would be to use more semantically valid controls such as buttons that submit the form. For example:
> 
> ```
> <form>
>   <button type="submit" name="authenticator" value="google-authenticator">
>     Google Authenticator
>   </button>
>   <button type="submit" name="authenticator" value="microsoft-authenticator">
>     Microsoft Authenticator
>   </button>
> </form>
> ```
> 
> This will submit the form with the appropriate `value` attached to the button when clicked. This also removes the need to have any JavaScript on the page. I've created [an example](https://stackblitz.com/edit/web-platform-6xfovm?file=index.html) to see how this works, let me know what you think.

I was not sure if div was a strong preference upstream or not. Buttons would be for the best indeed! I tried replacing div with button in the browser dev tools and some css adjustments would be necessary for the UI to remain the exact same after we change the tag to button.

**Comment by jonkoops** [comment]
> I was not sure if div was a strong preference upstream or not. Buttons would be for the best indeed! I tried replacing div with button in the browser dev tools and some css adjustments would be necessary for the UI to remain the exact same after we change the tag to button.

I think this was simply done at the time because the author was not aware this was an option, or did not want to deal with additional styling. Doing some CSS changes it totally fine, just be sure it works consistently across supported browser targets (Chrome, Firefox, Edge and Safari).

**Comment by Paul-Van-Uytvinck** [comment]
> > I was not sure if div was a strong preference upstream or not. Buttons would be for the best indeed! I tried replacing div with button in the browser dev tools and some css adjustments would be necessary for the UI to remain the exact same after we change the tag to button.
> 
> I think this was simply done at the time because the author was not aware this was an option, or did not want to deal with additional styling. Doing some CSS changes it totally fine, just be sure it works consistently across supported browser targets (Chrome, Firefox, Edge and Safari).

Tested in all 4 browsers

File: themes/src/main/resources/theme/base/login/webauthn-authenticate.ftl [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
                         <div class="${properties.kcFormClass!}">
                             <#list authenticators.authenticators as authenticator>
-                                <div id="kc-webauthn-authenticator" class="${properties.kcSelectAuthListItemClass!}">
+                                <button id="kc-webauthn-authenticator" class="${properties.kcSelectAuthListItemClass!}">
```
Comment (jonkoops) [inline themes/src/main/resources/theme/base/login/webauthn-authenticate.ftl:33]: Just realized these `id` attributes are likely not unique, as this is inside of a loop. We don't have to fix this now, so I logged a separate issue for it (#24462).

File: themes/src/main/resources/theme/base/login/select-authenticator.ftl [hunk 2]
```diff
@@ -32,7 +32,7 @@
                         <div class="${properties.kcSelectAuthListItemArrowClass!}">
                             <i class="${properties.kcSelectAuthListItemArrowIconClass!}"></i>
                         </div>
-                    </div>
+                    </button>
                 </#list>
                 <input type="hidden" id="authexec-hidden-input" name="authenticationExecution" />
```
Comment (jonkoops) [inline themes/src/main/resources/theme/base/login/select-authenticator.ftl:37]: If the aforementioned changes are made this hidden input can be safely removed.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: themes/src/main/resources/theme/base/login/webauthn-authenticate.ftl (hunk 1 above)
[the remark this answers is not in the document]
Comment (Paul-Van-Uytvinck) [reply themes/src/main/resources/theme/base/login/webauthn-authenticate.ftl:33]: Ah you are right, I thought we could click the key when several configured. I will remove it

**Review by jonkoops — APPROVED** [review]
LGTM, let's see what the tests do.

**Comment by jonkoops** [comment]
@Paul-Van-Uytvinck we can get this merged, can you squash your commits and [sign off on them](https://github.com/keycloak/keycloak/pull/24416/checks?check_run_id=18739116935)?

**Comment by Paul-Van-Uytvinck** [comment]
> @Paul-Van-Uytvinck we can get this merged, can you squash your commits and [sign off on them](https://github.com/keycloak/keycloak/pull/24416/checks?check_run_id=18739116935)?

I added my Signed-off-by line to every commit in this branch. We can squash the moment we merge, no? Or did you mean squash all commits locally?

**Review by ahus1 — APPROVED** [review]
Approving as of @jonkoops's review. Thank you for making the front end more accessible!

**Pull request merged**